Userland/Libraries/LibJS/Runtime/RegExpObject.cpp

```diff
@@ -100,53 +100,91 @@ RegExpObject::RegExpObject(String pattern, String flags, Object& prototype)
     }
 }
 
-void RegExpObject::initialize(GlobalObject& global_object)
+RegExpObject::~RegExpObject()
 {
-    auto& vm = this->vm();
-    Object::initialize(global_object);
-
-    define_native_property(vm.names.lastIndex, last_index, set_last_index, Attribute::Writable);
 }
 
-RegExpObject::~RegExpObject()
+// NON-STANDARD: Used to return the value of the ephemeral lastIndex property
```

Comment:
Why not add the property the right way and just sync it whenever you call into LibRegex?

Reply:
That also works, i just tried to emulate the exact behaviour of the native property implementation, because i didnt want to dig too deep into how the JS regex stuff works :sweat_smile: ill try doing that.

Reply:
done :)